### PR TITLE
Added detection for ZTE zxdsl devices

### PR DESCRIPTION
**Pull request by rsys-dev**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Hi @rsys-dev 

Thanx for your contribution. 
- Do you have a reason to create 2 YAML files instead of one for these devices ? I would suggest to merge them into a single one which would detect all the devices of the same family (they most probably behave exactly the same). 
- Could you, when this is done, add test data as shown in next post ? 

Thanx

**Comment by PipoCanaja** [comment]
And please also only push files linked to your change. All the .gitignore should be removed from this PR. Only your YAML files must appear here. Thanx

**Comment by rsys-dev** [comment]
@PipoCanaja 
I did this in two files to display the different Models in LibreNMS, but if this is unwanted, I try to change this to one file and display only "ZTE ZXDSL". I will add the requested tests asap.
The commit of .gitignore files was not intended, removed them, sorry.

**Comment by PipoCanaja** [comment]
@rsys-dev 
You can easily change the hardware description from the polling php files, exactly like serial number and software version. You can have a look at other OSes, they implement different solutions to gather the data (parsing sysDescr, snmpget on a specific OID, etc etc) 
/includes/polling/os/netgear.inc.php

**Comment by rsys-dev** [comment]
Hi,

I tried to run the tests but got this error on **./scripts/pre-commit.php -p -u**

```
1) LibreNMS\Tests\CommonTrapTest::testGenericTrap
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    'device_id' => 882
-    'message' => 'SNMP trap received: SNMPv2-MIB::someOid'
-    'type' => 'trap'
+    'device_id' => 0
+    'message' => 'Device 127.1.6.1 has been removed'
+    'type' => 'system'
     'reference' => null
     'username' => ''
-    'severity' => 2
+    'severity' => 3
 )

```
First I thougt there happened something with the Database, but even after cleanup and clean install this error persist.
Did I miss anything or has anybody an idea how to solve this?
127.1.6.1 is the device generated by 

`./scripts/save-test-data.php -o zxdsl`

**Comment by PipoCanaja** [comment]
Please update the code in the PR to implement only one OS instead of 2 right now, and add the snmprec file that was generated using collect-snmp-data.php, and the json fil generated by save-test-data.php. 
From that we can check if it goes well (and Travis, the automatic test running in each PR, will also check).

File: includes/polling/os/zxdsl.inc.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+use LibreNMS\RRD\RrdDefinition;
```
Comment (PipoCanaja) [inline includes/polling/os/zxdsl.inc.php]: You don't even need this statement here. There is no RRD work done in this file.

File: includes/polling/os/zxdsl.inc.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+if (strstr($device['sysObjectID'], '.1.3.6.1.4.1.3902.')) {
+    $sysObjectId = str_replace('.1.3.6.1.4.1.3902.', '', $device['sysObjectID']);
+    $sysObjectIdSplit = explode('.', $sysObjectId);
+    if (count($sysObjectIdSplit) >= 2) {
+        $hardware = "ZXDSL ".$sysObjectIdSplit[1];
+    }
```
Comment (PipoCanaja) [inline includes/polling/os/zxdsl.inc.php]: This is a way to get the model if (and only if) ZTE is consistant in the ObjectID and model numbers. But I would be surprised they don't have a dedicated OID that provides the hardware model and variant. You can snmpwalk the device completely and search for the model name in the resulting file.

Comment (rsys-dev) [reply includes/polling/os/zxdsl.inc.php]: Sure, this depends on the consistancy between ObjectID and Model, but there is no other place in the whole snmpwalk containing the model (ZXDSL9816 and 9806, have no others to check)
I will address this question to ZTE if there is another way to get the model.

File: includes/polling/os/zxdsl.inc.php (hunk 1 above)
Comment (rsys-dev) [reply includes/polling/os/zxdsl.inc.php]: Forgotten to remove this line, fixed

**Comment by PipoCanaja** [comment]
Good. Seems to be fairly good. 
Now, the next steps, if it makes sense with this device, could be to add more sensors. Could be fanspeed, temperature, anything that the vendor exposes in the MIBs. 
You can also gather the OS version, etc etc. The best way is to explore the ```snmpwalk -v2c -c <community> <deviceIP> . ``` and identify all data that you could expose in LibreNMS

File: includes/polling/os/zxdsl.inc.php (hunk 2 above)
Comment (rsys-dev) [reply includes/polling/os/zxdsl.inc.php]: They told me that the ObjectID is the correct way to find the model number.
Also there is an additional character in the ObjectID.
First example the ObjectID ends with "9806.22.1.1.1"
9806 -> Model
22 -> Additional character, 22. character of the latin alphabet
1.1.1 -> End of the Model
The model here is ZXDSL 9806V

Second example, the ObjectID ends with "9816.1.1.1"
9806 -> Model
1.1.1 -> End of the Model
The model here is ZXDSL 9816

I will add this in my next commit

**Comment by PipoCanaja** [comment]
Looks good. Could you generate again the JSON file and commit it ? The one that is in this PR now is not showing the right data, and TravisCI tests are not OK.

**Comment by rsys-dev** [comment]
> 
> 
> Looks good. Could you generate again the JSON file and commit it ? The one that is in this PR now is not showing the right data, and TravisCI tests are not OK.

Updated the JSON file, did not generate it after changing the limits. Sorry
All checks have passed now

**Comment by PipoCanaja** [comment]
No worry, looks good now. You can monitor the TravisCi tests after each commit so you find out this kind of issue.

**Pull request merged**